tests/e2e/050_nat_test.go

```diff
@@ -0,0 +1,516 @@
+		tcpSvcLocalPort = 8000
+		udpSvcLabel     = "udp-service"
+		udpSvcExtIP     = "80.80.80.50"
+		udpSvcExtPort   = 9999
+		udpSvcLocalPort = 9000
+
+		netMask = "/24"
+		ms1Name = "microservice1"
+		ms2Name = "microservice2"
+	)
+
+	vppTap1 := &vpp_interfaces.Interface{
+		Name:        vppTap1Name,
+		Type:        vpp_interfaces.Interface_TAP,
+		Enabled:     true,
+		IpAddresses: []string{vppTap1IP + netMask},
+		Link: &vpp_interfaces.Interface_Tap{
+			Tap: &vpp_interfaces.TapLink{
+				Version:        2,
+				ToMicroservice: msNamePrefix + ms1Name,
+			},
+		},
+	}
+	linuxTap1 := &linux_interfaces.Interface{
+		Name:        linuxTap1Name,
+		Type:        linux_interfaces.Interface_TAP_TO_VPP,
+		Enabled:     true,
+		IpAddresses: []string{linuxTap1IP + netMask},
+		HostIfName:  linuxTap1Hostname,
+		Link: &linux_interfaces.Interface_Tap{
+			Tap: &linux_interfaces.TapLink{
+				VppTapIfName: vppTap1Name,
+			},
+		},
+		Namespace: &linux_namespace.NetNamespace{
+			Type:      linux_namespace.NetNamespace_MICROSERVICE,
+			Reference: msNamePrefix + ms1Name,
+		},
+	}
+
+	vppTap2 := &vpp_interfaces.Interface{
+		Name:        vppTap2Name,
+		Type:        vpp_interfaces.Interface_TAP,
+		Enabled:     true,
+		IpAddresses: []string{vppTap2IP + netMask},
+		Link: &vpp_interfaces.Interface_Tap{
+			Tap: &vpp_interfaces.TapLink{
+				Version:        2,
+				ToMicroservice: msNamePrefix + ms2Name,
+			},
+		},
+	}
+	linuxTap2 := &linux_interfaces.Interface{
+		Name:        linuxTap2Name,
+		Type:        linux_interfaces.Interface_TAP_TO_VPP,
+		Enabled:     true,
+		IpAddresses: []string{linuxTap2IP + netMask},
+		HostIfName:  linuxTap2Hostname,
+		Link: &linux_interfaces.Interface_Tap{
+			Tap: &linux_interfaces.TapLink{
+				VppTapIfName: vppTap2Name,
+			},
+		},
+		Namespace: &linux_namespace.NetNamespace{
+			Type:      linux_namespace.NetNamespace_MICROSERVICE,
+			Reference: msNamePrefix + ms2Name,
+		},
+	}
+
+	ms2DefaultRoute := &linux_l3.Route{
+		OutgoingInterface: linuxTap2Name,
+		Scope:             linux_l3.Route_GLOBAL,
+		DstNetwork:        "0.0.0.0/0",
+		GwAddr:            vppTap2IP,
+	}
+
+	natGlobal := &vpp_nat.Nat44Global{
+		Forwarding: true,
+		NatInterfaces: []*vpp_nat.Nat44Global_Interface{
+			{
+				Name:     vppTap1Name,
+				IsInside: false,
+			},
+			{
+				Name:     vppTap2Name,
+				IsInside: true,
+			},
+		},
+	}
+
+	showTCPSvc := fmt.Sprintf("tcp local %s:%d external %s:%d vrf 0  out2in-only",
+		linuxTap2IP, tcpSvcLocalPort, tcpSvcExtIP, tcpSvcExtPort)
+	tcpSvc := &vpp_nat.DNat44{
+		Label: tcpSvcLabel,
+		StMappings: []*vpp_nat.DNat44_StaticMapping{
+			{
+				ExternalIp:   tcpSvcExtIP,
+				ExternalPort: tcpSvcExtPort,
+				Protocol:     vpp_nat.DNat44_TCP,
+				LocalIps: []*vpp_nat.DNat44_StaticMapping_LocalIP{
+					{
+						LocalIp:   linuxTap2IP,
+						LocalPort: tcpSvcLocalPort,
+					},
+				},
+			},
+		},
+	}
+
+	showUDPSvc := fmt.Sprintf("udp local %s:%d external %s:%d vrf 0  out2in-only",
```

Comment:
This should be closer to where it's used, because if the test fails there you will have to come here anyway.